data_safe_haven/commands/allowlist.py

```diff
@@ -0,0 +1,199 @@
+        typer.Option(help="File path to write the allowlist to."),
+    ] = None,
+) -> None:
+    """Print the current package allowlist"""
+    logger = get_logger()
+
+    try:
+        context = ContextManager.from_file().assert_context()
+    except DataSafeHavenConfigError as exc:
+        logger.critical(
+            "No context is selected. Use `dsh context add` to create a context "
+            "or `dsh context switch` to select one."
+        )
+        raise typer.Exit(1) from exc
+
+    sre_config = SREConfig.from_remote_by_name(context, name)
+
+    # Load Pulumi config
+    pulumi_config = DSHPulumiConfig.from_remote(context)
+
+    if sre_config.name not in pulumi_config.project_names:
+        msg = f"Could not load Pulumi settings for '{sre_config.name}'. Have you deployed the SRE?"
+        logger.error(msg)
+        raise typer.Exit(1)
+
+    sre_stack = SREProjectManager(
+        context=context,
+        config=sre_config,
+        pulumi_config=pulumi_config,
+    )
+
+    allowlist_storage_account = sre_stack.output("data")[
+        "storage_account_data_configuration_name"
+    ]
+
+    try:
+        allow_list = Allowlist.from_remote(
+            context=context,
+            repository=repository,
+            storage_account_name=allowlist_storage_account,
+            sre_resource_group=f"{sre_stack.stack_name}-rg",
+        )
+    except DataSafeHavenError as exc:
+        logger.critical(
+            "No allowlist is configured. Use `dsh allowlist upload` to create one."
+        )
+        raise typer.Exit(1) from exc
+
+    if file:
+        with open(file, "w") as f:
+            f.write(allow_list)
+    else:
+        console.print(allow_list)
+
+
+@allowlist_command_group.command()
+def template(
+    repository: Annotated[
+        AllowlistRepository,
+        typer.Argument(help="Name of the repository to show the allowlist for."),
+    ],
+    file: Annotated[
+        Optional[Path],  # noqa: UP007
+        typer.Option(help="File path to write allowlist template to."),
+    ] = None,
+) -> None:
+    """Print a template for the package allowlist"""
+
+    template_path = Path(
+        "data_safe_haven/resources",
+        "software_repositories",
+        "allowlists",
+        f"{repository.value}.allowlist",
+    )
+    with open(template_path) as f:
+        example_allowlist = f.read()
+    if file:
+        with open(file, "w") as f:
+            f.write(example_allowlist)
+        raise typer.Exit()
+    else:
+        console.print(example_allowlist)
+
+
+@allowlist_command_group.command()
+def upload(
+    name: Annotated[
+        str,
+        typer.Argument(help="Name of SRE to upload the allowlist for."),
+    ],
+    file: Annotated[
+        Path,
+        typer.Argument(help="Path to the allowlist file to upload."),
+    ],
+    repository: Annotated[
+        AllowlistRepository,
+        typer.Argument(help="Name of the repository to upload the allowlist for."),
```

Comment:
```suggestion
        typer.Argument(help="Repository type of the allowlist."),
```

I think this might be clearer. When you run `dsh allowlist upload --help` does it show all of the possible options?

Reply:
It does show the options, yes:

```
Usage: dsh allowlist upload [OPTIONS] NAME FILE REPOSITORY:{cran|pypi} 

Upload a package allowlist                                                                                                                                  
                                                                                                                                                             
╭─ Arguments ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ *    name            TEXT                    Name of SRE to upload the allowlist for. [default: None] [required]                                          │
│ *    file            PATH                    Path to the allowlist file to upload. [default: None] [required]                                             │
│ *    repository      REPOSITORY:{cran|pypi}  Name of the repository to upload the allowlist for. [default: None] [required]    
```